api-2.2/src/test/java/org/openmrs/module/fhir2/api/dao/impl/FhirConditionDaoImpl_2_2Test.java

```diff
@@ -90,14 +116,166 @@ public void shouldReturnNullWhenGetConditionByWrongUuid() {
 		assertThat(condition, nullValue());
 	}
 	
+	public void searchForPatients_shouldReturnConditionByPatientUuid() {
+		ReferenceParam patientReference = new ReferenceParam("", PATIENT_UUID);
+		ReferenceAndListParam patientList = new ReferenceAndListParam();
+		patientList.addValue(new ReferenceOrListParam().add(patientReference));
+		Collection<Condition> results = dao.searchForConditions(patientList, null, null, null, null, null, null, null);
+		
+		assertThat(results, notNullValue());
+		assertThat(results, not(empty()));
+		assertThat(results.size(), equalTo(3));
+		assertThat(results.iterator().next().getPatient().getUuid(), equalTo(PATIENT_UUID));
+	}
+	
+	@Test
+	public void searchForPatients_shouldReturnConditionByPatientGivenName() {
+		ReferenceParam patientReference = new ReferenceParam(Patient.SP_GIVEN, PATIENT_GIVEN_NAME);
+		ReferenceAndListParam patientList = new ReferenceAndListParam();
+		patientList.addValue(new ReferenceOrListParam().add(patientReference));
+		Collection<Condition> results = dao.searchForConditions(patientList, null, null, null, null, null, null, null);
+		
+		assertThat(results, notNullValue());
+		assertThat(results, not(empty()));
+		// TODO during review: Is returning duplicated Conditions the desired outcome?
+		assertThat(results.size(), equalTo(6));
```

Comment:
> Is returning duplicated Conditions the desired outcome?

Short version: no, we really shouldn't be getting duplicated conditions. Can you say more about why this is happening?

Reply:
Yep, that's a good point and probably out of scope for this ticket. I'll create a new ticket to resolve that issue. Small note on coding style, I prefer using the Hamcrest `hasSize()` method instead of `size(), equalTo()` as it results in a more meaningful error message when it fails.